src/EnergyPlus/DataSurfaces.cc

```diff
@@ -428,85 +428,208 @@ namespace DataSurfaces {
     Array2D<Real64> ReflFacBmToDiffSolObs;
     Array2D<Real64> ReflFacBmToDiffSolGnd;
     Array2D<Real64> ReflFacBmToBmSolObs;
     Array1D<Real64> ReflFacSkySolObs;
     Array1D<Real64> ReflFacSkySolGnd;
     Array2D<Real64> CosIncAveBmToBmSolObs;
-    Array1D<Real64> DBZoneIntWin; // Value of factor for beam solar entering a zone through interior windows
-    // (considered to contribute to diffuse in zone)
-    Array1D<Real64> SurfSunlitArea; // Sunlit area by surface number
-    Array1D<Real64> SurfSunlitFrac; // Sunlit fraction by surface number
-    // energy
-    Array1D<Real64> WinTransSolarEnergy; // Energy of WinTransSolar [J]
-    Array1D<Real64> WinBmSolarEnergy;    // Energy of WinBmSolar [J]
-
-    Array1D<Real64> WinBmBmSolarEnergy;  // Beam-to-beam energy of WinBmSolar [J]
-    Array1D<Real64> WinBmDifSolarEnergy; // Beam-to-diffuse energy of WinBmSolar [J]
-
-    Array1D<Real64> WinDifSolarEnergy;             // Energy of WinDifSolar [J]
-    Array1D<Real64> WinHeatGainRepEnergy;          // Energy of WinHeatGainRep [J]
-    Array1D<Real64> WinHeatLossRepEnergy;          // Energy of WinHeatLossRep [J]
-    Array1D<Real64> WinShadingAbsorbedSolarEnergy; // Energy of WinShadingAbsorbedSolar [J]
-    Array1D<Real64> WinGapConvHtFlowRepEnergy;     // Energy of WinGapConvHtFlowRep [J]
-    Array1D<Real64> WinHeatTransferRepEnergy;      // Energy of WinHeatTransfer [J]
+    Array1D<Real64> DBZoneIntWin;                          // Value of factor for beam solar entering a zone through interior windows (considered to contribute to diffuse in zone)
+    Array1D<Real64> SurfSunlitArea;                        // Sunlit area by surface number
+    Array1D<Real64> SurfSunlitFrac;                        // Sunlit fraction by surface number
+    Array1D<Real64> SurfSkySolarInc;                       // Incident diffuse solar from sky; if CalcSolRefl is true, includes reflection of sky diffuse and beam solar from exterior obstructions [W/m2]
+    Array1D<Real64> SurfGndSolarInc;                       // Incident diffuse solar from ground; if CalcSolRefl is true, accounts for shadowing of ground by building and obstructions [W/m2]
 
     std::vector<int> AllHTSurfaceList;          // List of all heat transfer surfaces
     std::vector<int> AllIZSurfaceList;          // List of all interzone heat transfer surfaces
     std::vector<int> AllHTNonWindowSurfaceList; // List of all non-window heat transfer surfaces
     std::vector<int> AllHTWindowSurfaceList;    // List of all window surfaces
     std::vector<int> AllSurfaceListReportOrder; // List of all surfaces - output reporting order
 
+    // Surface Window Heat Balance
+    Array1D<Real64> SurfWinTransSolar;                    // Exterior beam plus diffuse solar transmitted through window, or window plus shade/blind, into zone (W)
```

Comment:
This is a lot.  I know this is one of the parts of EnergyPlus where things are looped over quickly, so I concede that this provides a performance benefit, but just no one get any ideas about doing this in lots of places in the code where we don't do tight loops over array items.

Reply:
I didn't quite get the sentence after "but". By "doing this in lots of places in the code where we don't do tight loops over array items " do you mean using scalar arrays instead of struct arrays in places with loops that only call a few fields in the original SurfaceWindow struct?

Reply:
Yes, what is "this" in "this is a lot" referring to?

Reply:
If what you are referring to is the change from arrays of structs to arrays of scalars, then I don't think it's *only* a matter of tight loops. Tight loops (i.e., loops with no function calls or conditionals) are what you want for vectorization. But even without vectorization (which I don't think we are getting now) switching to arrays of scalars will improve your cache locality and utilization. If you have structs with a lot of fields that are only used in some calculations, in a given calculation you are probably bringing in a lot of that data unnecessarily. I think this is primarily what we are seeing here. I don't know that this type of transformation (which is really only partial for now) would be effective elsewhere in the code, but I would not dismiss it out of hand.